### ISSUE TITLE
UnitTest: add assertThrow, assertNoThrow

### ISSUE THREAD

**Issue by brianlheim** [supercollider/supercollider#3800]
1. UnitTest should have an `assertThrow` method that checks whether or not a given function throws an error. It should also require (or strongly encourage) the user to specify a class which should match the class of the thrown error.

2. Similarly, provide an `assertNoThrow` which checks that a given function does not throw.

Example signatures:

```supercollider
assertThrow { | func, errClass, message | /* ... */ }
assertNoThrow { | func, message | /* ... */ }
```

Example usage:

```supercollider
test_throw {
    this.assertThrow( { 5.meow }, DoesNotUnderstandError, "An unknown message should throw an error");
}

test_noThrow {
    this.assertNoThrow( { 5.squared }, "Normal usage should not throw");
}
```

**Comment by telephon** [comment]
Very good idea, and very good direction. I think it helps to make unit testing more readable and simpler to write.

One question: when a throw happens in a normal unit test method, shouldn't this somehow lead to a fail?

**Comment by telephon** [comment]
I just noticed that

```supercollider
try { \stone.throw } { "stone was thrown" }; 
```

never calls the second function. This is intended, I think. It is just that perhaps we want to call the `assertThrow` method perhaps `assertException`?

Otherwise we need to implement a `catch`, which we currently don't have (or need, as it seems).

**Comment by telephon** [comment]
Here is a draft: #3801

**Comment by brianlheim** [comment]
Fixed in #3808 - thanks @telephon !

**Comment by telephon** [comment]
thank you @brianlheim !
